psst-gui/src/webapi/client.rs

```diff
@@ -452,44 +779,153 @@ impl WebApi {
     }
 
     // https://developer.spotify.com/documentation/web-api/reference/save-shows-user
     pub fn save_show(&self, id: &str) -> Result<(), Error> {
-        let request = self.put("v1/me/shows")?.query("ids", id);
+        let request = self.put("v1/me/shows", None)?.query("ids", id);
         self.send_empty_json(request)?;
         Ok(())
     }
 
     // https://developer.spotify.com/documentation/web-api/reference/remove-shows-user
     pub fn unsave_show(&self, id: &str) -> Result<(), Error> {
-        let request = self.delete("v1/me/shows")?.query("ids", id);
+        let request = self.delete("v1/me/shows", None)?.query("ids", id);
         self.send_empty_json(request)?;
         Ok(())
     }
 }
 
+
+/* 
+impl WebApi {
+    // https://developer.spotify.com/documentation/web-api/reference/search/
+    pub fn search(
+        &self,
+        query: &str,
+        topics: &[SearchTopic],
+        limit: usize,
+    ) -> Result<SearchResults, Error> {
+        #[derive(Deserialize)]
+        struct ApiSearchResults {
+            artists: Option<Page<Artist>>,
+            albums: Option<Page<Arc<Album>>>,
+            tracks: Option<Page<Arc<Track>>>,
+            playlists: Option<Page<Playlist>>,
+            shows: Option<Page<Arc<Show>>>,
+        }
+
+        let topics = topics.iter().map(SearchTopic::as_str).join(",");
+        let request = self
+            .get("v1/search")?
+            .query("q", query)
+            .query("type", &topics)
+            .query("limit", &limit.to_string())
+            .query("marker", "from_token");
+        let result: ApiSearchResults = self.load(request)?;
+
+        let artists = result.artists.map_or_else(Vector::new, |page| page.items);
+        let albums = result.albums.map_or_else(Vector::new, |page| page.items);
+        let tracks = result.tracks.map_or_else(Vector::new, |page| page.items);
+        let playlists = result.playlists.map_or_else(Vector::new, |page| page.items);
+        let shows = result.shows.map_or_else(Vector::new, |page| page.items);
+        Ok(SearchResults {
+            query: query.into(),
+            artists,
+            albums,
+            tracks,
+            playlists,
+            shows,
+        })
+    }
+}
+*/
 /// View endpoints.
 impl WebApi {
-    pub fn get_made_for_you(&self) -> Result<Vector<Playlist>, Error> {
+    fn build_home_request(&self, section_uri: &str) -> (String, String) {
+        let extensions = json!({
+            "persistedQuery": {
+                "version": 1,
+                "sha256Hash": "4da53a78e4e98d4f3fa55698af5b751fe05ca3a1a4a526ff8147e8866ccfa49f"
+            }
+        });
+
+        let variables = json!( {
+            "uri": section_uri,
+            "timeZone": "Europe/London",
+            "sp_t": self.access_token().unwrap(),  // Assuming this returns a Result<String, Error>
+            "country": "GB",
+            "sectionItemsOffset": 0,
+            "sectionItemsLimit": 20,
+        });
+
```

Comment:
We should be getting this from the user.

Reply:
Yes I agree!